python/dgl/sparse/sparse_matrix.py

```diff
@@ -479,6 +479,74 @@ def is_diag(self):
         """Returns whether the sparse matrix is a diagonal matrix."""
         return self.c_sparse_matrix.is_diag()
 
+    def select(self, dim: int, ids: torch.Tensor or slice):
+        """Returns a sub-matrix selected by a list of ids or by range.
+
+        Parameters
+        ----------
+        dim : int
+            The dim to select from matrix, should be 0 or 1.
+        ids : tensor.Tensor or slice
+            Tensor indicates the row ids to select from matrix.
+            NOTE: The ids can include duplicate values.
+            Slice indicates sub-matrix selected by range [start, end).
+
+        The function has not supported autograd.
+
+        Returns
+        -------
+        SparseMatrix
+            The sub-matrix only contains rows or columns selected.
+
+        Examples
+        --------
+
+        Case1: Select several rows by index.
+
+        >>> indices = torch.tensor([0, 1, 1, 2, 3, 4], [0, 2, 4, 3, 5, 0]])
+        >>> val = torch.tensor([0, 1, 2, 3, 4, 5])
+        >>> A = dglsp.spmatrix(indices, val)
+        >>> row_ids = torch.tensor([0, 1, 4])
+        >>> A.select(0, row_ids)
+        SparseMatrix(indices=tensor([[0, 1, 1, 2],
+                                     [0, 2, 4, 0]]),
+                     values=tensor([0, 1, 2, 5]),
+                     shape=(3, 6), nnz=4)
+
+        Case2: Select several columns by index.
+
+        >>> column_ids = torch.tensor([0, 4, 5])
+        >>> A.select(1, column_ids)
+        SparseMatrix(indices=tensor([[0, 1, 3, 4],
+                                     [0, 1, 2, 0]]),
+                     values=tensor([0, 2, 4, 5]),
+                     shape=(5, 3), nnz=4)
+
+        Case3: Select several rows by range.
+
+        >>> A.select(0, slice(1, 3))
+        SparseMatrix(indices=tensor([[0, 0, 1],
+                                     [2, 4, 3]]),
+                     values=tensor([1, 2, 3]),
+                     shape=(2, 6), nnz=3)
+
+        Case4: Select several columns by range.
+
+        >>> A.select(1, slice(3, 6))
+        SparseMatrix(indices=tensor([[1, 2, 3],
+                                     [1, 0, 2]]),
+                     values=tensor([2, 3, 4]),
+                     shape=(5, 3), nnz=3)
+        """
+        if isinstance(ids, torch.Tensor):
+            raise NotImplementedError
+        elif isinstance(ids, slice):
+            start = 0 if ids.start is None else ids.start
+            end = ids.stop
+            raise NotImplementedError
+        else:
+            raise TypeError
```

Comment:
Add more info about the TypeError, see example:
https://github.com/dmlc/dgl/blob/44a9faad7ab60d6bbcc08e094c0eaa86a0d73063/python/dgl/graphbolt/itemset.py#L67

Reply:
Fixed.